### PR TITLE
Fix some docs phrasings and headers

### PR DESCRIPTION
**Pull request by strickvl**
...as mentioned in a livecoding session just now...

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/developer-guide/runtime-configuration.md [hunk 1]
```diff
@@ -9,7 +9,14 @@ Business logic is what defines
 a step and the pipeline. Step and pipeline configurations are used to
 dynamically set parameters at runtime.
 
-## Step configuration
+You can configure your pipelines at runtime in the following ways:
+
+* Configure from within the code: Do this when you are quickly iterating on your code 
+and don't wait to your actual step code. This is useful in the development phase.
```
Comment (strickvl) [inline docs/book/developer-guide/runtime-configuration.md:15]:
```suggestion
and don't want to change your actual step code. This is useful in the development phase.
```

Possibly? The original phrasing was unclear for me.

File: docs/book/introduction/core-concepts.md [hunk 2]
```diff
@@ -11,10 +11,11 @@ when using ZenML, starting with the most basic to things you'll only encounter
 when deploying your work to the cloud. At the very highest level, the workflow
 is as follows:
 
-- You write your code to define what you want to happen in your machine learning
+- You write your code as a pipeline to define what you want to happen in your machine learning
   workflow
 - You configure a ZenML Stack which is the infrastructure and setup that will
-  run your machine learning code
+  run your machine learning
```
Comment (htahir1) [inline docs/book/introduction/core-concepts.md:17]:
```suggestion
  run your machine learning code
```

File: docs/book/developer-guide/materializer.md [hunk 3]
```diff
@@ -77,7 +77,7 @@ Each materializer has `ASSOCIATED_TYPES` and `ASSOCIATED_ARTIFACT_TYPES`.
   etc. This is simply a tag to query certain artifact types in the
   post-execution workflow.
 
-### Extending the `BaseMaterializer`
+# Writing a custom materializer
```
Comment (htahir1) [inline docs/book/developer-guide/materializer.md:80]: IMO only the main headline should be h1 and I would use h2 for all subheaders within a page like this one. Might reformatting this whole file tho.

**Review by htahir1 — CHANGES_REQUESTED** [review]
Great changes. Especially moving the run name section makes a lot of sense. Overall I would push to have the headline usage somewhat standardized across pages, i.e. where to use h1, h2, ... This still seems a bit inconsistent.

- Felix fromHamza's machine

**Comment by htahir1** [comment]
I added @fa9r as a reviewer as you made this review from my PC :D. @AlexejPenner maybe also take a look?

**Comment by htahir1** [comment]
@AlexejPenner  and @fa9r please do take a look when you have time

**Review by fa9r — APPROVED** [review]
LGTM now :)

**Review by AlexejPenner — APPROVED** [review]
:rocket:

**Pull request merged**